src/state/apis/nft/nftApi.ts

```diff
@@ -170,34 +172,41 @@ export const nftApi = createApi({
     getNftCollection: build.query<NftCollectionType, GetNftCollectionInput>({
       queryFn: async ({ collectionId, accountIds }, { dispatch }) => {
         try {
-          const { assetReference: collectionAddress, chainId } = fromAssetId(collectionId)
-          const alchemyCollectionData = await getAlchemyInstanceByChainId(chainId)
-            .nft.getContractMetadata(collectionAddress)
-            .then(contract => parseAlchemyNftContractToCollectionItem(contract, chainId))
-
-          // Alchemy is the most/only reliable source for collection data for now
-          if (alchemyCollectionData) {
-            dispatch(nft.actions.upsertCollection(alchemyCollectionData))
-            return { data: alchemyCollectionData }
-          }
-
-          // Note, this will consistently fail, as getZapperCollectionBalance is monkey patched for the RTK query to reject
-          // The reason for that is Zapper is currently rugged upstream - if we don't get Alchemy data, we're out of luck and can't get meta
-          const { data: zapperCollectionData } = await dispatch(
-            zapperApi.endpoints.getZapperCollectionBalance.initiate({ accountIds, collectionId }),
+          const services = [
+            getAlchemyCollectionData,
+            (collectionId: AssetId, accountIds: AccountId[]) =>
+              dispatch(
+                zapperApi.endpoints.getZapperCollectionBalance.initiate({
+                  accountIds,
+                  collectionId,
+                }),
+              ),
+          ]
+
+          const results = await Promise.all(
+            services.map(service => service(collectionId, accountIds)),
           )
 
-          if (!zapperCollectionData)
-            return {
-              error: {
-                status: 404,
-                data: {
-                  message: 'Collection not found',
-                },
-              },
+          const collectionData = results.reduce<NftCollectionType | null>((acc, result) => {
+            const { data } = result
+
+            if (!data) return acc
+
+            if (!acc) {
+              acc = data
+            } else {
+              acc = updateNftCollection(acc, data)
             }
 
-          return { data: zapperCollectionData }
+            return acc
```

Comment:
prefer we return the desired acc rather than mutate, e.g (cant add suggestion here stupid github)

```
if (!acc) return data
return updateNftCollection(acc, data)
```

Reply:
Abso-lutely 0aceb187da6a08d5ea0b4d726caaded414293981